src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3641,17 +3641,54 @@ void addToStringMethod(MethodSymbol printMethod)
 
                 if (!memberSignatures.TryGetValue(targetMethod, out Symbol? existingToStringMethod))
                 {
-                    var toStringMethod = new SynthesizedRecordToString(this, printMethod, memberOffset: members.Count, diagnostics);
-                    members.Add(toStringMethod);
+                    var baseToStringMethod = getBaseToStringMethod();
+                    switch (baseToStringMethod)
+                    {
+                        case null:
+                        case { IsSealed: false }:
+                            var toStringMethod = new SynthesizedRecordToString(this, printMethod, memberOffset: members.Count, diagnostics);
+                            members.Add(toStringMethod);
+                            break;
+                        case { IsSealed: true }:
+                            if (!this.DeclaringCompilation.IsFeatureEnabled(MessageID.IDS_FeatureSealedToStringInRecord))
+                            {
+                                diagnostics.Add(ErrorCode.ERR_InheritingFromRecordWithSealedToString, this.Locations[0]);
+                            }
+                            break;
+                    }
                 }
                 else
                 {
                     var toStringMethod = (MethodSymbol)existingToStringMethod;
                     if (!SynthesizedRecordObjectMethod.VerifyOverridesMethodFromObject(toStringMethod, SpecialMember.System_Object__ToString, diagnostics) && toStringMethod.IsSealed && !IsSealed)
                     {
-                        diagnostics.Add(ErrorCode.ERR_SealedAPIInRecord, toStringMethod.Locations[0], toStringMethod);
+                        MessageID.IDS_FeatureSealedToStringInRecord.CheckFeatureAvailability(
+                            diagnostics,
+                            this.DeclaringCompilation,
+                            toStringMethod.Locations[0]);
                     }
                 }
+
+                MethodSymbol? getBaseToStringMethod()
+                {
+                    var objectToString = this.DeclaringCompilation.GetSpecialTypeMember(SpecialMember.System_Object__ToString);
+                    var tmp = this.BaseTypeNoUseSiteDiagnostics;
```

Comment:
>tmp [](start = 24, length = 3)

Please use more descriptive name for the local

Reply:
Done